src/Access/BlockAccessCheck.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Drupal\tide_landing_page\Access;
+
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Routing\Access\AccessInterface;
+
+/**
+ * Determines access for block layout based on roles.
+ */
+class BlockAccessCheck implements AccessInterface {
+
+  /**
+   * Checks access.
+   *
+   * @param Drupal\Core\Session\AccountInterface $account
+   *   The currently logged in account.
+   *
+   * @return bool
+   *   A \Drupal\Core\Access\AccessInterface constant value.
+   */
+  public function access(AccountInterface $account) {
+    $rolesWithAccessRestricted = [
```

Comment:
Hi @yeniatencio , please avoid doing internal value assignment, which makes the function untestable.

Reply:
Ticket required to check for restricting access based on roles.

Reply:
I think this may arbitrarily block(or unblock) all users to access the `/admin/structure/block` page, as the codes break the `administer blocks` permission that associated with the route, and this also affects other modules that may have custom permissions.

Reply:
@vincent-gao , I have updated the PR description. Please read above.